### PR TITLE
Improvements to the Java Library plugin docs

### PR DESCRIPTION
**Pull request by melix**
This pull request is a follow up to the Java Library documentation one.

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/javaLibraryPlugin.xml [hunk 1]
```diff
@@ -128,9 +123,13 @@
             </listitem>
             <listitem>The configurations in <firstterm>blue</firstterm> are internal to the component, for its own use
             </listitem>
-            <listitem>The configurations in <firstterm>white</firstterm> are legacy configurations that can be used to declare dependencies too
+            <listitem>The configurations in <firstterm>white</firstterm> are configurations inherited from the Java plugin
             </listitem>
         </itemizedlist>
+        <note>
+            The <firstterm>compile</firstterm>, <firstterm>testCompile</firstterm>, <firstterm>runtime</firstterm> and <firstterm>testRuntime</firstterm> configurations
+            inherited from the Java plugin are still available but are deprecated. You should avoid using them, as they are only keptfor backwards compatibility.
```
Comment (jjohannes) [inline subprojects/docs/src/docs/userguide/javaLibraryPlugin.xml]: Missing a space: `keptfor` -> `kept for`

File: subprojects/docs/src/docs/userguide/javaLibraryPlugin.xml [hunk 2]
```diff
@@ -87,29 +87,24 @@
             <listitem>types exclusively found in internal classes (future versions of Gradle will let you declare which packages belong to the public API)</listitem>
```
Comment (jjohannes) [inline subprojects/docs/src/docs/userguide/javaLibraryPlugin.xml]: Reading this again, I am wondering if we should mention (recommend?) that if your want to have a good separation within your own (large) multi-project build, you should organise it into "api" and "implementation" sub-projects.

Maybe not needed for this initial guide though...

Comment (melix) [inline subprojects/docs/src/docs/userguide/javaLibraryPlugin.xml]: I don't think it makes sense. Projects have internal classes, and we should not force users to separate what they want to expose from what they don't want to expose. It's orthogonal to this API/impl separation.

Comment (jjohannes) [inline subprojects/docs/src/docs/userguide/javaLibraryPlugin.xml]: ok

**Pull request merged**